Userland/Libraries/LibCore/Promise.h

```diff
@@ -20,21 +27,35 @@ class Promise : public Object {
     void resolve(Result&& result)
     {
         m_pending = move(result);
+        m_state = PromiseState::Resolved;
+
         if (on_resolved)
             on_resolved(m_pending.value());
     }
 
-    bool is_resolved()
+    void cancel()
```

Comment:
Background action has an on_rejected or on_error callback. Why did we not adopt the same behavior here?

When someone cancels or rejects a promise, the person responsible for the rejection or cancellation should be responsible for providing the Error associated with that.

```
void cancel(Error error) {
   m_state = cancelled;
   m_rejection_reason = move(error);
}
```

Reply:
That would probably require redesigning a lot of  users

Reply:
What users? You just added the cancel API in this PR.

Reply:
Sure, I mirrored the Promise API. Whatever

Reply:
This comment is on the Promise API. `Core::Promise::cancel()` was added in this commit. Or do you mean you were mirroring a Promise API from another language or library?